src/data/finset/basic.lean

```diff
@@ -1938,6 +1939,26 @@ ext $ λ ⟨x, y⟩, by simp only [mem_product, mem_bind, mem_image, exists_prop
 @[simp] theorem card_product (s : finset α) (t : finset β) : card (s.product t) = card s * card t :=
 multiset.card_product _ _
 
+theorem filter_product (p : α → Prop) (q : β → Prop) [decidable_pred p] [decidable_pred q] :
+  (s.product t).filter (λ (x : α × β), p x.1 ∧ q x.2) = (s.filter p).product (t.filter q) :=
+by { ext ⟨a, b⟩, simp only [mem_filter, mem_product], finish, }
+
+instance (p : α → Prop) (q : β → Prop) [decidable_pred p] [decidable_pred q] :
+  decidable_pred (λ (x : α × β), p x.1 = q x.2) :=
+by { rintros ⟨a, b⟩, simp only [eq_iff_iff], apply_instance, }
```

Comment:
This belongs somewhere in `src/logic`.

Reply:
Thanks, now moved.

In fact I just realised something even more basic seems to be missing, namely:
```lean
instance [decidable a] [decidable b] : decidable (a = b) := ...
```
So I've added this instead.

Reply:
Well the preference is that you should use iff instead of eq for equality of propositions. Where did this come up?